### PR TITLE
refactoring, java version update

### PR DESCRIPTION
**Pull request by antsher**
updated java compatibility version to be compatible with Java 8 so runnables can be replaced with lambdas, updated some libs' dependencies, also fixed some annotations when overriding methods, replaced deprecated FirebaseCrash with Crashlytics and added a needed WAKE_LOCK permission for FirebaseAnalytics

### CODE REVIEW

File: app/src/main/java/ch/sheremet/katarina/cocktailspro/model/Ingredients.java [hunk 1]
```diff
@@ -111,9 +114,10 @@ public void setBeverageId(String mBeverageId) {
         this.mBeverageId = mBeverageId;
     }
 
+    @NonNull
     @Override
     public String toString() {
-        return String.format("Ingredients{mId=%d, mIngredient='%s', mMeasure='%s', mBeverageId='%s'}",
+        return String.format(Locale.getDefault(), "Ingredients{mId=%d, mIngredient='%s', mMeasure='%s', mBeverageId='%s'}",
```
Comment (ksheremet) [inline app/src/main/java/ch/sheremet/katarina/cocktailspro/model/Ingredients.java:120]: Regarding doc: https://docs.oracle.com/javase/8/docs/api/java/util/Formatter.html Optional locale as the first argument can be used to get locale-specific formatting of numbers. In this case we have only strings that came from server. I assume they won't be formated. WDYT?

**Review by ksheremet — APPROVED** [review]

File: app/src/main/java/ch/sheremet/katarina/cocktailspro/model/Ingredients.java (hunk 1 above)
Comment (antsher) [reply app/src/main/java/ch/sheremet/katarina/cocktailspro/model/Ingredients.java:120]: Well, actually mId is a number, so it is recommended to use a locale :)

Comment (ksheremet) [reply app/src/main/java/ch/sheremet/katarina/cocktailspro/model/Ingredients.java:120]: Yep, but it is an id :)

Comment (ksheremet) [reply app/src/main/java/ch/sheremet/katarina/cocktailspro/model/Ingredients.java:120]: But, anyway, thanks for Pull Request. Great work!

**Pull request merged**